npm/svelte/src/mount.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  injectStylesBeforeElement,
+  StyleOptions,
+  getContainerEl,
+  setupHooks,
+} from '@cypress/mount-utils'
+import { SvelteComponentTyped, ComponentConstructorOptions, SvelteComponent } from 'svelte'
+
+const DEFAULT_COMP_NAME = 'unknown'
+
+type SvelteComponentClass = typeof SvelteComponent
+type ConstructorOptions = Omit<ComponentConstructorOptions, 'hydrate' | 'target' | '$$inline'>
+
+export interface MountOptions extends ConstructorOptions, Partial<StyleOptions> {
+  log?: boolean
+}
+
+export interface MountReturn {
+  component: SvelteComponentTyped
+  unmount: (unmountOptions: { log?: boolean, message?: string }) => Cypress.Chainable<undefined>
+}
+
+let componentInstance: SvelteComponentTyped | undefined
+
+const cleanup = () => {
+  componentInstance?.$destroy()
+}
+
+// Extract the component name from the object passed to mount
+const getComponentDisplayName = (Component: SvelteComponentClass): string => {
+  if (Component.name) {
+    const [_, match] = /Proxy\<(\w+)\>/.exec(Component.name) || []
+
+    return match || Component.name
+  }
+
+  return DEFAULT_COMP_NAME
+}
+
+export function mount (
+  Component: SvelteComponentClass,
+  options: MountOptions = {},
+): Cypress.Chainable<MountReturn> {
+  return cy.then(() => {
+    const target = getContainerEl()
+
+    injectStylesBeforeElement(options, document, target)
+
+    const ComponentConstructor = ((Component as any).default || Component) as SvelteComponentClass
+
+    componentInstance = new ComponentConstructor({
+      target,
+      ...options,
+    })
+
+    const unmount: MountReturn['unmount'] = (unmountOptions = {}) => {
+      return cy.then(() => {
+        cleanup()
+        if (unmountOptions.log) {
+          Cypress.log({
+            name: 'unmount',
+            message: [unmountOptions.message ?? 'Unmounted component'],
+          })
+        }
+      })
+    }
+
+    return cy.wait(1, { log: false }).then(() => {
```

Comment:
Out of curiosity what's the bug introduced if we do not `wait(1)`? I think React does this too, I forgot why. Maybe a comment for ourselves in the future?

Reply:
Found [this comment](https://github.com/cypress-io/cypress/blob/da15c5acbcb89abfbba3d3a52dd05d4a86acecf4/npm/react/src/createMount.ts#L122), I'll add the comment in, best to be safe. Since it's all wrapped in a chainable I feel like it should be deferred but that could be the illusion that chainables look/act like promises.